client/state/utils/actions.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Internal dependencies
+ */
+import { UTILS_UNIQUEID_INCREMENT } from 'state/action-types';
+
+export function uniqueId( prefix ) {
+	return ( dispatch, getState ) => {
```

Comment:
@rralian if the store receives a function, the thunk middleware will run it instead of simply returning the action object. This is to allow things like asynchronous calls which shouldn't dispatch until the async part has resolved. That's why there's a `dispatch` parameter to the thunk here, because it is giving us the control to decided _when_ or _how_.

That being said, I think splitting the actual action creator would be a good idea.

```js
const responseAction = response => ( {
	type: RESPONSE_ACTION,
	response
} );

const asyncAction = dispatch => {
	getFromServer()
		.then( response => dispatch( responseAction( response ) ) )
		.catch( error.log );
};
```